edXRegressionAutomated/Tests/LMS/pages/courseware.py

```diff
@@ -0,0 +1,20 @@
+from bok_choy.page_object import PageObject
+
+class CoursewarePage(PageObject):
+    """
+    Courseware Page
+    """
+
+    url = None
+
+    def is_browser_on_page(self):
+        # Check if Login page appears then Login first
+
+        if 'log into' in self.browser.title:
```

Comment:
This kind of logic should not be in the `is_browser_on_page` method. They should be in a separate location as part of the test workflow. (So, in the test file, not the page file.) 

`is_browser_on_page` should be a quick way of verifying that you are on the page you promise to be on. It is run in the background almost continuously so it should be accurate, but also light.